client/webpack.config.js

```diff
@@ -113,6 +114,16 @@ function filterEntrypoints( entrypoints ) {
 	return allowed;
 }
 
+function addHotMiddlewareIfDev( entry ) {
+	if ( isDevelopment ) {
+		return Object.entries( entry ).reduce( ( acc, [ key, value ] ) => {
+			acc[ key ] = [ 'webpack-hot-middleware/client', ...value ];
+			return acc;
+		}, {} );
```

Comment:
Alternatively, we could wrap in a `Object.fromEntries()` which will do the job to reduce back to an object, and we can only do a `.map()` to change the value. Could be done as part of `filterEndpoints()`.

Reply:
I changed it to use fromEntries here: 4dd9489c. However, I am wondering if this code should even be here, see discussion below.